src/finch/interface/tensor.py

```diff
@@ -20,15 +20,283 @@ def to_numpy(self):
 
     @abstractmethod
     def __add__(self, other):
-        compute(lazy.lazy(self).__add__(other))
+        lazy.compute(lazy.lazy(self).__add__(other))
```

Comment:
For dunder binary ops (functions that take the "other" argument), we should return a lazy array if other is lazy. This allows lazytensor to propagate through the computation.